src/api/java/com/minecolonies/api/research/AbstractResearchProvider.java

```diff
@@ -532,42 +537,83 @@ public Research addResearchRequirement(final ResourceLocation researchReq, final
         }
 
         /**
-         * Adds an item cost to the research.  This will be consumed when beginning the research, and will not be refunded.
+         * Adds an item cost to the research. This will be consumed when beginning the research, and will not be refunded.
          * Multiple ItemCosts are supported, but for UI reasons it's encouraged to keep to 5 or less.
-         * @param item   The item stack to require, including count.
+         *
+         * @param item  The item to require.
+         * @param count The number of the item to require.
          * @return this.
          */
-        public Research addItemCost(final ItemStack item)
+        public Research addItemCost(final Item item, final int count)
         {
-            final JsonArray reqArray;
-            if(this.json.has("requirements") && this.json.get("requirements").isJsonArray())
-            {
-                reqArray = this.json.getAsJsonArray("requirements");
-                this.json.remove("requirements");
-            }
-            else
+            return addItemCost(List.of(item), count);
+        }
+
+        /**
+         * Adds an item cost to the research. This will be consumed when beginning the research, and will not be refunded.
+         * Multiple ItemCosts are supported, but for UI reasons it's encouraged to keep to 5 or less.
+         *
+         * @param items The item to require.
+         * @param count The number of the item to require.
+         * @return this.
+         */
+        public Research addItemCost(final List<Item> items, final int count)
```

Comment:
is there a reason we're using Items instead of ItemStacks?

Reply:
This previous ItemStack one was not being used. So I did not bother continuing it either, I only kept the item/count one and added the tag/list ones.